pkg/ocm/client/types.go

```diff
@@ -0,0 +1,73 @@
+package client
+
+import (
+	"context"
+	"net/http"
+)
+
+type SingleClusterSubResource[T any] interface {
```

Comment:
Just curious as to the naming convention, and what you think for sub resources that are a collection and have a list?